mayachemtools/docs/modules/man3/RotatableBondsDescriptors.3
.\" Automatically generated by Pod::Man 2.28 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "RotatableBondsDescriptors 1"
.TH RotatableBondsDescriptors 1 "2018-05-15" "perl v5.22.4" "MayaChemTools"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
RotatableBondsDescriptors
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
use MolecularDescriptors::RotatableBondsDescriptors;
.PP
use MolecularDescriptors::RotatableBondsDescriptors qw(:all);
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBRotatableBondsDescriptors\fR class provides the following methods:
.PP
new, GenerateDescriptors, GetDescriptorNames,
StringifyRotatableBondsDescriptors
.PP
\&\fBRotatableBondsDescriptors\fR is derived from \fBMolecularDescriptors\fR class which in turn
is  derived from \fBObjectProperty\fR base class that provides methods not explicitly defined
in \fBRotatableBondsDescriptors\fR, \fBMolecularDescriptors\fR or \fBObjectProperty\fR classes using Perl's
\&\s-1AUTOLOAD\s0 functionality. These methods are generated on-the-fly for a specified object property:
.PP
.Vb 3
\&    Set<PropertyName>(<PropertyValue>);
\&    $PropertyValue = Get<PropertyName>();
\&    Delete<PropertyName>();
.Ve
.PP
A rotatable bond [ Ref 92 ] is defined as any single bond which is not in a ring and involves only non-hydrogen
atoms. By default, the following types of single bonds are not considered rotatable bonds:
.PP
.Vb 5
\&    o Terminal bonds
\&    o Bonds attached to triple bonds
\&    o Amide C\-N bonds
\&    o Thioamide C\-N bond bonds
\&    o Sulfonamide S\-N bonds
.Ve
.SS "\s-1METHODS\s0"
.IX Subsection "METHODS"
.IP "\fBnew\fR" 4
.IX Item "new"
.Vb 3
\&    $RotatableBondsDescriptors = new MolecularDescriptors::
\&                                 RotatableBondsDescriptors(
\&                                 %NamesAndValues);
.Ve
.Sp
Using specified \fIRotatableBondsDescriptors\fR property names and values hash, \fBnew\fR
method creates a new object and returns a reference to newly created \fBRotatableBondsDescriptors\fR
object. By default, the following properties are initialized:
.Sp
.Vb 9
\&    Molecule = \*(Aq\*(Aq
\&    Type = \*(AqRotatableBonds\*(Aq
\&    IgnoreTerminalBonds = 1
\&    IgnoreBondsToTripleBonds = 1
\&    IgnoreAmideBonds = 1
\&    IgnoreThioamideBonds = 1
\&    IgnoreSulfonamideBonds = 1
\&    @DescriptorNames = (\*(AqRotatableBonds\*(Aq)
\&    @DescriptorValues = (\*(AqNone\*(Aq)
.Ve
.Sp
Examples:
.Sp
.Vb 2
\&    $RotatableBondsDescriptors = new MolecularDescriptors::
\&                                 RotatableBondsDescriptors();
\&
\&    $RotatableBondsDescriptors = new MolecularDescriptors::
\&                                 RotatableBondsDescriptors(
\&                                 \*(AqIgnoreAmideBonds\*(Aq => 0,
\&                                 \*(AqIgnoreThioamideBonds\*(Aq => 0,
\&                                 \*(AqIgnoreSulfonamideBonds\*(Aq => 0);
\&
\&    $RotatableBondsDescriptors\->SetMolecule($Molecule);
\&    $RotatableBondsDescriptors\->GenerateDescriptors();
\&    print "RotatableBondsDescriptors: $RotatableBondsDescriptors\en";
.Ve
.IP "\fBGenerateDescriptors\fR" 4
.IX Item "GenerateDescriptors"
.Vb 1
\&    $RotatableBondsDescriptors\->GenerateDescriptors();
.Ve
.Sp
Calculates number of rotatable bonds descriptors in a molecule and returns
\&\fIRotatableBondsDescriptors\fR.
.IP "\fBGetDescriptorNames\fR" 4
.IX Item "GetDescriptorNames"
.Vb 3
\&    @DescriptorNames = $RotatableBondsDescriptors\->GetDescriptorNames();
\&    @DescriptorNames = MolecularDescriptors::RotatableBondsDescriptors::
\&                       GetDescriptorNames();
.Ve
.Sp
Returns all available descriptor names as an array.
.IP "\fBStringifyRotatableBondsDescriptors\fR" 4
.IX Item "StringifyRotatableBondsDescriptors"
.Vb 1
\&    $String = $RotatableBondsDescriptors\->StringifyRotatableBondsDescriptors();
.Ve
.Sp
Returns a string containing information about \fIRotatableBondsDescriptors\fR object.
.SH "AUTHOR"
.IX Header "AUTHOR"
Manish Sud <msud@san.rr.com>
.SH "SEE ALSO"
.IX Header "SEE ALSO"
MolecularDescriptors.pm, MolecularDescriptorsGenerator.pm
.SH "COPYRIGHT"
.IX Header "COPYRIGHT"
Copyright (C) 2018 Manish Sud. All rights reserved.
.PP
This file is part of MayaChemTools.
.PP
MayaChemTools is free software; you can redistribute it and/or modify it under
the terms of the \s-1GNU\s0 Lesser General Public License as published by the Free
Software Foundation; either version 3 of the License, or (at your option)
any later version.


scripts/mayachemtools/lib/MolecularDescriptors/RotatableBondsDescriptors.pm
package MolecularDescriptors::RotatableBondsDescriptors;
#
# File: RotatableBondsDescriptors.pm
# Author: Manish Sud <msud@san.rr.com>
#
# Copyright (C) 2018 Manish Sud. All rights reserved.
#
# This file is part of MayaChemTools.
#
# MayaChemTools is free software; you can redistribute it and/or modify it under
# the terms of the GNU Lesser General Public License as published by the Free
# Software Foundation; either version 3 of the License, or (at your option) any
# later version.
#
# MayaChemTools is distributed in the hope that it will be useful, but without
# any warranty; without even the implied warranty of merchantability of fitness
# for a particular purpose.  See the GNU Lesser General Public License for more
# details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with MayaChemTools; if not, see <http://www.gnu.org/licenses/> or
# write to the Free Software Foundation Inc., 59 Temple Place, Suite 330,
# Boston, MA, 02111-1307, USA.
#

use strict;
use Carp;
use Exporter;
use Scalar::Util ();
use TextUtil ();
use Atom;
use Molecule;
use MolecularDescriptors::MolecularDescriptors;

use vars qw(@ISA @EXPORT @EXPORT_OK %EXPORT_TAGS);

@ISA = qw(MolecularDescriptors::MolecularDescriptors Exporter);
@EXPORT = qw();
@EXPORT_OK = qw(GetDescriptorNames);

%EXPORT_TAGS = (all  => [@EXPORT, @EXPORT_OK]);

# Setup class variables...
my($ClassName, @DescriptorNames);
_InitializeClass();

# Overload Perl functions...
use overload '""' => 'StringifyRotatableBondsDescriptors';

# Class constructor...
sub new {
  my($Class, %NamesAndValues) = @_;

  # Initialize object...
  my $This = $Class->SUPER::new();
  bless $This, ref($Class) || $Class;
  $This->_InitializeRotatableBondsDescriptors();

  $This->_InitializeRotatableBondsDescriptorsProperties(%NamesAndValues);

  return $This;
}

# Initialize class ...
sub _InitializeClass {
  #Class name...
  $ClassName = __PACKAGE__;

  # Descriptor names...
  @DescriptorNames = ('RotatableBonds');

}

# Get descriptor names as an array.
#
# This functionality can be either invoked as a class function or an
# object method.
#
sub GetDescriptorNames {
  return @DescriptorNames;
}

# Initialize object data...
#
sub _InitializeRotatableBondsDescriptors {
  my($This) = @_;

  # Type of MolecularDescriptor...
  $This->{Type} = 'RotatableBonds';

  # MayaChemTools rotatable bonds default definition corresponds to modifed
  # version of rotatable bonds definition used by Veber et al. [ Ref 92 ]
  #
  $This->{IgnoreTerminalBonds} = 1;
  $This->{IgnoreBondsToTripleBonds} = 1;
  $This->{IgnoreAmideBonds} = 1;
  $This->{IgnoreThioamideBonds} = 1;
  $This->{IgnoreSulfonamideBonds} = 1;

  # Intialize descriptor names and values...
  $This->_InitializeDescriptorNamesAndValues(@DescriptorNames);

  return $This;
}

# Initialize object properties...
#
sub _InitializeRotatableBondsDescriptorsProperties {
  my($This, %NamesAndValues) = @_;

  my($Name, $Value, $MethodName);
  while (($Name, $Value) = each  %NamesAndValues) {
    $MethodName = "Set${Name}";
    $This->$MethodName($Value);
  }

  return $This;
}

# Calculate number of rotatable bonds in a molecule...
#
# A rotatable bond is defined as any single bond which is not in a ring
# and involves only non-hydrogen atoms. By default, the following types
# of single bonds are not considered rotatable bonds:
#
#   . Terminal bonds
#   . Bonds attached to triple bonds
#   . Amide C-N bonds
#   . Thioamide C-N bond bonds
#   . Sulfonamide S-N bonds
#
# MayaChemTools rotatable bonds default definition corresponds to modifed
# version of rotatable bonds definition used by Veber et al. [ Ref 92 ]
#
sub GenerateDescriptors {
  my($This) = @_;

  # Initialize descriptor values...
  $This->_InitializeDescriptorValues();

  # Check availability of molecule...
  if (!$This->{Molecule}) {
    carp "Warning: ${ClassName}->GenerateDescriptors: $This->{Type} molecular descriptors generation didn't succeed: Molecule data is not available: Molecule object hasn't been set...";
    return undef;
  }

  # Calculate descriptor values...
  if (!$This->_CalculateDescriptorValues()) {
    carp "Warning: ${ClassName}->GenerateDescriptors: $This->{Type} molecular descriptors generation didn't succeed: Couldn't calculate RotatableBonds values...";
    return undef;
  }

  # Set final descriptor values...
  $This->_SetFinalDescriptorValues();

  return $This;
}

# Calculate RotatableBonds value...
#
sub _CalculateDescriptorValues {
  my($This) = @_;
  my($Bond, $RotatableBonds, $Atom1, $Atom2);

  $RotatableBonds = 0;

  BOND: for $Bond ($This->{Molecule}->GetBonds()) {
    # Is it a non-ring ring bond?
    if (!$Bond->IsSingle() || $Bond->IsInRing()) {
      next BOND;
    }

    ($Atom1, $Atom2) = $Bond->GetAtoms();

    # Does bond contain any Hydrogen atoms?
    if ($Atom1->IsHydrogen() || $Atom2->IsHydrogen()) {
      next BOND;
    }

    # Check for terminal bonds...
    if ($This->{IgnoreTerminalBonds} && $This->_IsTerminalBond($Atom1, $Atom2)) {
      next BOND;
    }

    # Check for bonds attached to triple bonds...
    if ($This->{IgnoreBondsToTripleBonds} && $This->_IsAttachedToTripleBond($Atom1, $Atom2)) {
      next BOND;
    }

    # Check for amide bonds...
    if ($This->{IgnoreAmideBonds} && $This->_IsAmideBond($Atom1, $Atom2)) {
      next BOND;
    }

    # Check for amide bonds...
    if ($This->{IgnoreThioamideBonds} && $This->_IsThioamideBond($Atom1, $Atom2)) {
      next BOND;
    }

    # Check for sulfonamide bonds...
    if ($This->{IgnoreSulfonamideBonds} && $This->_IsSulfonamideBond($Atom1, $Atom2)) {
      next BOND;
    }

    $RotatableBonds += 1;
  }

  # Track the calculated values...
  $This->{RotatableBonds} = $RotatableBonds;

  return $This;
}

# Is it a terminal bond?
#
sub _IsTerminalBond {
  my($This, $Atom1, $Atom2) = @_;

  return ($Atom1->GetAtomicInvariantValue('X') <= 1 || $Atom2->GetAtomicInvariantValue('X') <= 1 ) ? 1 : 0;
}

# Is it attached to a terminal bond?
#
sub _IsAttachedToTripleBond {
  my($This, $Atom1, $Atom2) = @_;

  return ($Atom1->GetAtomicInvariantValue('LBO') == 3 || $Atom2->GetAtomicInvariantValue('LBO') == 3) ? 1 : 0;
}

# Is it an amide bond?
#
# Amide: R-C(=O)-N(-R)(-R")
#
sub _IsAmideBond {
  my($This, $Atom1, $Atom2) = @_;
  my($CarbonAtom, $NitrogenAtom);

  ($CarbonAtom, $NitrogenAtom) = (undef, undef);

  if ($Atom1->IsCarbon() && $Atom2->IsNitrogen()) {
    ($CarbonAtom, $NitrogenAtom) = ($Atom1, $Atom2);
  }
  elsif ($Atom2->IsCarbon() && $Atom1->IsNitrogen()) {
    ($CarbonAtom, $NitrogenAtom) = ($Atom2, $Atom1);
  }

  if (!$CarbonAtom) {
    return 0;
  }

  return $CarbonAtom->DoesAtomNeighborhoodMatch('C.T3.DB1', ['O', 'N', 'C,H'], ['=', '-', '-']) ? 1 : 0;
}

# Is it a thioamide bond?
#
# Thioamide: R-C(=S)-N(-R)(-R")
#
sub _IsThioamideBond {
  my($This, $Atom1, $Atom2) = @_;
  my($CarbonAtom, $NitrogenAtom);

  ($CarbonAtom, $NitrogenAtom) = (undef, undef);

  if ($Atom1->IsCarbon() && $Atom2->IsNitrogen()) {
    ($CarbonAtom, $NitrogenAtom) = ($Atom1, $Atom2);
  }
  elsif ($Atom2->IsCarbon() && $Atom1->IsNitrogen()) {
    ($CarbonAtom, $NitrogenAtom) = ($Atom2, $Atom1);
  }

  if (!$CarbonAtom) {
    return 0;
  }

  return $CarbonAtom->DoesAtomNeighborhoodMatch('C.T3.DB1', ['S', 'N', 'C,H'], ['=', '-', '-']) ? 1 : 0;
}

# Is it a sulfonamide bond?
#
# Sulfonamide: R-S(=O)(=O)-N(-R)(-R")
#
sub _IsSulfonamideBond {
  my($This, $Atom1, $Atom2) = @_;
  my($SulfurAtom, $NitrogenAtom);

  ($SulfurAtom, $NitrogenAtom) = (undef, undef);

  if ($Atom1->IsSulfur() && $Atom2->IsNitrogen()) {
    ($SulfurAtom, $NitrogenAtom) = ($Atom1, $Atom2);
  }
  elsif ($Atom2->IsSulfur() && $Atom1->IsNitrogen()) {
    ($SulfurAtom, $NitrogenAtom) = ($Atom2, $Atom1);
  }

  if (!$SulfurAtom) {
    return 0;
  }

  return $SulfurAtom->DoesAtomNeighborhoodMatch('S.T4.DB2', ['O', 'O', 'N', '!O'], ['=', '=', '-', '-']) ? 1 : 0;
}

# Setup final descriptor values...
#
sub _SetFinalDescriptorValues {
  my($This) = @_;

  $This->{DescriptorsGenerated} = 1;

  $This->SetDescriptorValues($This->{RotatableBonds});

  return $This;
}

# Return a string containg data for RotatableBondsDescriptors object...
#
sub StringifyRotatableBondsDescriptors {
  my($This) = @_;
  my($RotatableBondsDescriptorsString);

  # Type of MolecularDescriptors...
  $RotatableBondsDescriptorsString = "MolecularDescriptorType: $This->{Type}; IgnoreTerminalBonds: " . ($This->{IgnoreTerminalBonds} ? "Yes" : "No") . "; IgnoreBondsToTripleBonds: " .  ($This->{IgnoreBondsToTripleBonds} ? "Yes" : "No") . "; IgnoreAmideBonds: " .  ($This->{IgnoreAmideBonds} ? "Yes" : "No") . "; IgnoreThioamideBonds: " .  ($This->{IgnoreThioamideBonds} ? "Yes" : "No") . "; IgnoreSulfonamideBonds: " .  ($This->{IgnoreSulfonamideBonds} ? "Yes" : "No");

  # Setup molecular descriptor information...
  $RotatableBondsDescriptorsString .= "; " . $This->_StringifyDescriptorNamesAndValues();

  return $RotatableBondsDescriptorsString;
}

# Is it a RotatableBondsDescriptors object?
sub _IsRotatableBondsDescriptors {
  my($Object) = @_;

  return (Scalar::Util::blessed($Object) && $Object->isa($ClassName)) ? 1 : 0;
}

1;

__END__

=head1 NAME

RotatableBondsDescriptors

=head1 SYNOPSIS

use MolecularDescriptors::RotatableBondsDescriptors;

use MolecularDescriptors::RotatableBondsDescriptors qw(:all);

=head1 DESCRIPTION

B<RotatableBondsDescriptors> class provides the following methods:

new, GenerateDescriptors, GetDescriptorNames,
StringifyRotatableBondsDescriptors

B<RotatableBondsDescriptors> is derived from B<MolecularDescriptors> class which in turn
is  derived from B<ObjectProperty> base class that provides methods not explicitly defined
in B<RotatableBondsDescriptors>, B<MolecularDescriptors> or B<ObjectProperty> classes using Perl's
AUTOLOAD functionality. These methods are generated on-the-fly for a specified object property:

    Set<PropertyName>(<PropertyValue>);
    $PropertyValue = Get<PropertyName>();
    Delete<PropertyName>();

A rotatable bond [ Ref 92 ] is defined as any single bond which is not in a ring and involves only non-hydrogen
atoms. By default, the following types of single bonds are not considered rotatable bonds:

    o Terminal bonds
    o Bonds attached to triple bonds
    o Amide C-N bonds
    o Thioamide C-N bond bonds
    o Sulfonamide S-N bonds

=head2 METHODS

=over 4

=item B<new>

    $RotatableBondsDescriptors = new MolecularDescriptors::
                                 RotatableBondsDescriptors(
                                 %NamesAndValues);

Using specified I<RotatableBondsDescriptors> property names and values hash, B<new>
method creates a new object and returns a reference to newly created B<RotatableBondsDescriptors>
object. By default, the following properties are initialized:

    Molecule = ''
    Type = 'RotatableBonds'
    IgnoreTerminalBonds = 1
    IgnoreBondsToTripleBonds = 1
    IgnoreAmideBonds = 1
    IgnoreThioamideBonds = 1
    IgnoreSulfonamideBonds = 1
    @DescriptorNames = ('RotatableBonds')
    @DescriptorValues = ('None')

Examples:

    $RotatableBondsDescriptors = new MolecularDescriptors::
                                 RotatableBondsDescriptors();

    $RotatableBondsDescriptors = new MolecularDescriptors::
                                 RotatableBondsDescriptors(
                                 'IgnoreAmideBonds' => 0,
                                 'IgnoreThioamideBonds' => 0,
                                 'IgnoreSulfonamideBonds' => 0);

    $RotatableBondsDescriptors->SetMolecule($Molecule);
    $RotatableBondsDescriptors->GenerateDescriptors();
    print "RotatableBondsDescriptors: $RotatableBondsDescriptors\n";

=item B<GenerateDescriptors>

    $RotatableBondsDescriptors->GenerateDescriptors();

Calculates number of rotatable bonds descriptors in a molecule and returns
I<RotatableBondsDescriptors>.

=item B<GetDescriptorNames>

    @DescriptorNames = $RotatableBondsDescriptors->GetDescriptorNames();
    @DescriptorNames = MolecularDescriptors::RotatableBondsDescriptors::
                       GetDescriptorNames();

Returns all available descriptor names as an array.

=item B<StringifyRotatableBondsDescriptors>

    $String = $RotatableBondsDescriptors->StringifyRotatableBondsDescriptors();

Returns a string containing information about I<RotatableBondsDescriptors> object.

=back

=head1 AUTHOR

Manish Sud <msud@san.rr.com>

=head1 SEE ALSO

MolecularDescriptors.pm, MolecularDescriptorsGenerator.pm

=head1 COPYRIGHT

Copyright (C) 2018 Manish Sud. All rights reserved.

This file is part of MayaChemTools.

MayaChemTools is free software; you can redistribute it and/or modify it under
the terms of the GNU Lesser General Public License as published by the Free
Software Foundation; either version 3 of the License, or (at your option)
any later version.

=cut
